lib/udf_string_utils.rb

```diff
@@ -244,6 +244,55 @@ class UdfStringUtils
                            {query: "select ?(1, 3, 1, ',')", expect: "1,2" , example: true},
                            {query: "select ?(1, 5, 2, '|')", expect: "1|3", example: true},
                        ]
+      }, {
+          type:        :function,
```

Comment:
There is a lot of duplication between two UDFs. I would just create one UDF (e.g. `parse_room_description`) that takes two parameters:

* `component`: a string with the component you want the UDF to return. E.g. `adults`, `children`, `rooms`.
* `room`: the room description

I would also have that order for the parameters so emulate Redshift built-in functions such as `date_part` or `date_trunc`.

And a general comment. I wouldn't add this utility to the `udf_string_utils.rb` module. Instead I would create a new module and call it `udf_findhotel.rb` which we could use for utilities that are very specific to FindHotel.